backend/src/utils/email_send.rs

```diff
@@ -0,0 +1,224 @@
+use chrono::Utc;
+use crypto::digest::Digest;
+use crypto::hmac::Hmac;
+use crypto::mac::Mac;
+use crypto::sha2::Sha256;
+use hex;
+use lazy_static::lazy_static;
+use reqwest::header::HeaderMap;
+use serde::Serialize;
+use std::env;
+
+lazy_static! {
+    static ref SECRET_ID: String = env::var("SECRET_ID").ok().unwrap_or_else(|| "".to_string());
+    static ref SECRET_KEY: String = env::var("SECRET_KEY")
+        .ok()
+        .unwrap_or_else(|| "".to_string());
+}
+
+#[derive(Serialize)]
+struct Template {
+    #[serde(rename = "TemplateID")]
+    template_id: i32,
+    #[serde(rename = "TemplateData")]
+    template_data: String,
+}
+
+#[derive(Serialize)]
+struct Body {
+    #[serde(rename = "FromEmailAddress")]
+    from_email_address: String,
+    #[serde(rename = "Destination")]
+    destination: Vec<String>,
+    #[serde(rename = "Template")]
+    template: Template,
+    #[serde(rename = "Subject")]
+    subject: String,
+}
+
+fn hash(s: String) -> String {
+    let mut hasher = Sha256::new();
+    hasher.input_str(&s);
+    hasher.result_str()
+}
+
+fn gen_payload(param: &Body) -> String {
+    let payload_str = serde_json::to_string(param).unwrap();
+    let payload_vec: Vec<&str> = payload_str.split(':').collect();
+    let payload = payload_vec.join(": ");
+    let payload_vec: Vec<&str> = payload.split(',').collect();
+    payload_vec.join(", ")
+}
+
+fn sign<'a>(key: &'a [u8], msg: &'a [u8]) -> Vec<u8> {
+    let mut hmac = Hmac::new(Sha256::new(), key);
+    hmac.input(msg);
+    let result = hmac.result();
+    let code = result.code();
+    code.to_vec()
+}
+
+fn assemble_headers(timestamp: String) -> HeaderMap {
```

Comment:
Can also test the function result like that:

![image](https://user-images.githubusercontent.com/44333669/146517766-7142d031-ebcf-4865-8f3b-00040da0ca46.png)

![image](https://user-images.githubusercontent.com/44333669/146517868-0cf5cef8-c5ee-4af4-bfb5-8286f59dd88c.png)